### PR TITLE
Fix bug causing `recordsByDirectory` to fail when file names contain "_"

### PR DESCRIPTION
**Pull request by ewanv**
Bug was caused by using a file glob as a regex.

### CODE REVIEW

File: core/src/test/scala/au/com/cba/omnia/thermometer/example/ExecutionSpec.scala [hunk 1]
```diff
@@ -107,4 +108,20 @@ Demonstration of ThermometerSpec using Execution monad
       })
     )
   }
+
+  def underscoreFiles = withEnvironment(path(getClass.getResource("env4").toString)) {
+    val file = new File(s"$dir/user/output/cars/underscore_file")
```
Comment (toddmowen) [inline core/src/test/scala/au/com/cba/omnia/thermometer/example/ExecutionSpec.scala]: Shouldn't there be a call `executesOk(execution2)` like the other similar tests?

Comment (ewanv) [inline core/src/test/scala/au/com/cba/omnia/thermometer/example/ExecutionSpec.scala]: Given that this test is only concerned about files containing an underscore, using an `Execution` isn't necessary. I wasn't able to find an easy way to specify the output file name using a `TypedPipe`.

File: core/src/test/scala/au/com/cba/omnia/thermometer/example/ExecutionSpec.scala [hunk 2]
```diff
@@ -107,4 +108,20 @@ Demonstration of ThermometerSpec using Execution monad
...
+    file.getParentFile.mkdirs
+    file.createNewFile
+
+    val pw = new PrintWriter(file)
+    pw.append("Batmobile|1966|NOT_IMP")
```
Comment (toddmowen) [inline core/src/test/scala/au/com/cba/omnia/thermometer/example/ExecutionSpec.scala]: The underscore in `NOT_IMP` might be confusing. It's nothing to do with the test, but as a reader I immediately wondered if it was.

File: core/src/test/scala/au/com/cba/omnia/thermometer/example/ExecutionSpec.scala (hunk 1 above)
Comment (toddmowen) [inline core/src/test/scala/au/com/cba/omnia/thermometer/example/ExecutionSpec.scala]: ok

File: core/src/test/scala/au/com/cba/omnia/thermometer/example/ExecutionSpec.scala (hunk 2 above)
Comment (ewanv) [inline core/src/test/scala/au/com/cba/omnia/thermometer/example/ExecutionSpec.scala]: Good point. I was just copying the data from the expected files, but "DUMMY" is probably more appropriate.

**Pull request merged**